### PR TITLE
Feature toggle visibility widgets

### PR DESCRIPTION
**Pull request by Bluesmile82**
- Add toggleVisibility button to preview in Analysis page
- Add visible attribute to widget
- Hide the graph in the page when visible is on
- Add a sample user 
- Update readme in the setup

### CODE REVIEW

File: app/assets/javascripts/views/shared/chartWidgetView.js [hunk 1]
```diff
@@ -253,7 +255,36 @@
...
+     * Render the Toggle Visibility button and attach an event listener to it
+     */
+    _renderToggleVisibilityButton: function () {
+      // We create the button
+      var button = document.createElement('button');
+      button.type = 'button';
+      button.classList.add('c-button', 'toggle-visibility-button');
+      button.textContent = 'Hide';
+      // We attach the listener
+      button.addEventListener('click', () => {
```
Comment (sorodrigo) [inline app/assets/javascripts/views/shared/chartWidgetView.js]: Avoid using ES6 syntax (i.e. arrow functions) cause Backbone projects are not transpiled.

File: app/assets/javascripts/views/shared/chartWidgetView.js [hunk 2]
```diff
@@ -253,7 +255,36 @@
...
+        this.options.toggleVisibilityCallback();
+        this._renderChart();
+      });
 
+      // We append the button to the DOM
+      this.el.querySelector('.js-toggle-visibility-button').appendChild(button);
+    },
+    _toggleWidgetVisibility: function(){
+      const visibleButton = this.el.querySelector('.toggle-visibility-button');
+      const largeWidget = visibleButton.closest('.large-widget');
```
Comment (sorodrigo) [inline app/assets/javascripts/views/shared/chartWidgetView.js]: `closest` method is not supported in all major browsers, use a different selector or fallback to jQuery. We're already using it in other views in this project.

File: app/assets/stylesheets/layouts/shared/l-analysis-dashboard.scss [hunk 3]
```diff
@@ -90,6 +90,10 @@ $secondary-container-initial-width: 400px;
         // Please make them coincide
         min-height: #{$media-max-width * .5};
       }
+
+      &.-is-not-visible{
+        min-height: 0;
```
Comment (sorodrigo) [inline app/assets/stylesheets/layouts/shared/l-analysis-dashboard.scss]: `hidden` or `isHidden` is better. Also don't hide elements using the height attribute.

**Comment by tiagojsag** [comment]
@Bluesmile82 I still don't see the charts/maps. Did you fix the issue @sorodrigo reported?

**Comment by sorodrigo** [comment]
@tiagojsag I'm working on this PR will push changes soon

**Comment by tiagojsag** [comment]
The bottom right chart is only visible if the bottom left chart is also enabled.

File: README.md [hunk 4]
```diff
@@ -35,6 +35,7 @@ To set up the database, run:
     bundle exec rake db:create
     bundle exec rake db:migrate
     bundle exec rake db:seed
+    bundle exec rake db:site_settings:update
```
Comment (tiagojsag) [inline README.md]: this is only needed if you are updating from a previous version.

**Comment by tiagojsag** [comment]
please add an entry to the changelog

File: app/assets/javascripts/routers/management/PreviewStepRouter.js [hunk 5]
```diff
@@ -125,10 +125,11 @@
     },
     /**
      * Save the state of the specified component into a global state object
-     * @param {string} component - "filters", "chart1", "chart2" or "map"
+     * @param {string} component - "filters", "widget1", "widget2"
      * @param {object} state - state to save
      */
     _saveState: function (component, state) {
+      debugger;
```
Comment (tiagojsag) [inline app/assets/javascripts/routers/management/PreviewStepRouter.js]: @sorodrigo debugger on the code => you buy beers for everyone!

**Comment by tiagojsag** [comment]
The hide/show eye icon has a 1 pixel vertical position difference. Can you fix this please?
thanks

**Pull request merged**